packages/docprovider-extension/src/index.ts

```diff
@@ -17,14 +17,19 @@ import {
   WebSocketProviderWithLocks
 } from '@jupyterlab/docprovider';
 import { ServerConnection } from '@jupyterlab/services';
+import { IStateDB } from '@jupyterlab/statedb';
 
 /**
  * The default document provider plugin
  */
 const docProviderPlugin: JupyterFrontEndPlugin<IDocumentProviderFactory> = {
   id: '@jupyterlab/docprovider-extension:plugin',
+  optional: [IStateDB],
   provides: IDocumentProviderFactory,
-  activate: (app: JupyterFrontEnd): IDocumentProviderFactory => {
+  activate: (
+    app: JupyterFrontEnd,
+    state: IStateDB
```

Comment:
We should also add `| null` since `IStateDB` is optional (to get the strick null checks):

```suggestion
    state: IStateDB | null
```